cpp/src/arrow/compute/exec/filter_node.cc

```diff
@@ -103,17 +103,9 @@ class FilterNode : public MapNode {
   }
 
   void InputReceived(ExecNode* input, ExecBatch batch) override {
-    EVENT(span_, "InputReceived", {{"batch.length", batch.length}});
```

Comment:
Does this need a `TraceInputReceived` or `NoteInputReceived`?

Reply:
It gets it from `MapNode`.  This will be more obvious in https://github.com/apache/arrow/pull/15253 because subclasses will no longer implement `InputReceived` (they will use `MapNode::InputReceived` and instead just implement `ProcessBatch`)